archive_updater.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import subprocess
+import os
+
+MAX_NUMBER_OF_RECENT_COMMITS = 2
+    
+def fill_message_directory(archive_path: str, directory_path: str, last_used_commit_hash: str) -> str:
+    '''Given a git repo path, directory path, and last used commit hash, this method updates the git repo,
+    then retrieves the MAX_NUMBER_OF_RECENT_COMMITS recent commits and converts them into files stored in
+    the directory corresponding to the passed in directory path. Throws CalledProcessError and Exception'''
+    
+    fetch = subprocess.check_call(['git', '-C', archive_path, 'fetch'])
+    
+    log = subprocess.Popen(
+        ['git', '-C', archive_path, 'log', '--format=format:%H', '-n',str(MAX_NUMBER_OF_RECENT_COMMITS)],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE)
```

Comment:
is the reason for the use of Popen() to get stderr and stdout?

See https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError
> **stderr**
> Stderr output of the child process if it was captured by run(). Otherwise, None.

can just use one of the check_* funcs